ft-logic/src/lib.rs

```diff
@@ -316,25 +316,45 @@ impl FTLogic {
     }
 }
 
-#[gstd::async_main]
-async fn main() {
-    let action: FTLogicAction = msg::load().expect("Error in loading `StorageAction`");
-    let logic: &mut FTLogic = unsafe { FT_LOGIC.get_or_insert(Default::default()) };
-    match action {
-        FTLogicAction::Message {
-            transaction_hash,
-            account,
-            payload,
-        } => logic.message(transaction_hash, &account, &payload).await,
-        FTLogicAction::UpdateStorageCodeHash(storage_code_hash) => {
-            logic.update_storage_hash(storage_code_hash)
+fn __main_safe() {
```

Comment:
Change to `async fn main`

Reply:
it's being called from extern "C" so how it's possible to be async?